romtools/workflows/residual_evaluator.py

```diff
@@ -0,0 +1,184 @@
+"""
+Protocol for interfacing with external application to compute residual snapshots
+corresponding to existing state snapshots.
+"""
+
+from typing import Protocol, Iterable
+import numpy as np
+
+from romtools.workflows.workflow_utils import create_empty_dir
+
+
+class SteadyResidualEvaluator(Protocol):
+    """
+    Baseline steady residual evaluator protocol
+    """
+
+    def load_projected_full_solution(self, filename: str) -> np.ndarray:
+        """
+        Reads the full-order model solution from the specified filename
+
+        Args:
+            filename (str): filename of the file containing the full-order model solution data
+
+        Returns:
+            `np.ndarray`: The projected full-order solution in a 1-dimensional array
+
+        """
+        pass
```

Comment:
i don't understand this method: is this supposed to load the already projected solution or it is supposed to load the fom solution and then project it?

Reply:
It's supposed to load the FOM and then project it, so a potentially better name is "compute_reduced_state"